### ISSUE TITLE
Retrive more values with `search` subcommand

### ISSUE THREAD

**Issue by ramiuslr** [marevers/pleasant-cli#2]
Hi, thanks for this amazing tool it helps me a lot at work !

I would like to know if it is possible to get more attributes while searching using `pleasant-cli search`.
At this time it looks like we have the following fields returned:
```console
GroupId
Id
Name
Notes
Path
Tags
Url
Username
```

For example I need a specific field `.Credentials[].CustomUserFields.Connect` to be also listed in the final result.

Would this be possible ?
Feel free to ask if you need more context or precision I'll be happy to give you any other information you may need.

**Comment by marevers** [comment]
Hey @ramiuslr , thanks for opening this issue.

I need to have a look at what the search API returns. Right now I am only retaining certain fields from the results but I can look into if it's possible to return more.

Give me some time to investigate and I will let you know if it's possible to implement those custom user fields.

**Comment by ramiuslr** [comment]
Awesome, thank you so much !
I have very little experience in Go but feel free to ask if I can help in any way !

**Comment by marevers** [comment]
@ramiuslr Unfortunately, this is not possible with just the `search` command on its own - have a look at the API documentation here: [API v5 SearchResults](https://pleasantpasswords.com/info/pleasant-password-server/m-programmatic-access/restful-api/restful-api-v5/service-models-v5#SearchResults). The entry results are limited to the ones you already mentioned.

I am not sure in what way exactly you are using the CLI, but what you could do (in a script for example) is the following:

1. Retrieve the entries you need using `pleasant-cli search`
2. Loop over the `Credentials` array and execute a `pleasant-cli get entry --id <id>` for each of them, using the id of each entry

That should output the custom user fields with their respective values.

Please let me know if that helps. If not, we can see if there is another solution.

**Comment by ramiuslr** [comment]
Thank you for your quick reply !

In fact I'm not sure it will work like this, just because I would like to search based on some extra attributes.

I mean for example if I need to look for an entry which has `192.168.1.100` in his field `.Credentials[].CustomUserFields.Connect`, I guess the only option would be to retrieve the whole list of entries in the server, then loop on each one to see if his custom field matches the query.

However we have a lot of entries on our server so that would be so long and inefficient.
I understand this is a limitation for *Pleasant* API, so probably no solution here unfortunately :smile:

**Comment by ramiuslr** [comment]
Here are just a few more precision about my use case.

I have two wrapper scripts I put in my path.
The first one is to search and display formatted output to terminal:
```bash
#!/bin/env bash
pleasant-cli search --pretty -q "${1}" | \
  jq -r '.Credentials[] | "\nID:\t\(.Id)\nPATH:\t\(.Path)\nURL:\t\(.Url)\nNAME:\t\(.Name)\nLOGIN:\t\(.Username)\n"'
```
Output in terminal is a list of the following entries:
```console
ID:	<ID>
PATH:	pleasant/folder/of/entry
URL:	https://example.com
NAME:	Example entry name
LOGIN:	username
```

Then I copy to my clipboard the ID of the entry I need.

The second one is an autotype script and is triggered by a keyboard shortcut (I'm using *Wayland*):
```bash
#!/bin/env bash
echo "type $(pleasant-cli get entry -i $(wl-paste) | jq -r '.Username')" | DOTOOL_XKB_LAYOUT=fr dotool
echo "key tab" | DOTOOL_XKB_LAYOUT=fr dotool
echo "type $(pleasant-cli get entry --password -i $(wl-paste))" | DOTOOL_XKB_LAYOUT=fr dotool
echo "key enter" | DOTOOL_XKB_LAYOUT=fr dotool
```

I use autotype because there are some cases where I can't paste (*Teamviewer* for example).

**Comment by marevers** [comment]
> I mean for example if I need to look for an entry which has `192.168.1.100` in his field `.Credentials[].CustomUserFields.Connect`, I guess the only option would be to retrieve the whole list of entries in the server, then loop on the list to see if each entry has a custom field that matches the query.

Understood. Yes, now I understand the usecase a little better. You would like to search and find only entries based on the contents of the CustomUserFields. The query API indeed does not support that seeing as it the search results are only based upon the fields visible in the results. The CustomerUserFields are not considered within that.

**Comment by ramiuslr** [comment]
Yes I totally agree, as we unfortunately cannot do anything from the client side for now, we can close this. Looking forward for a new feature in the Pleasant API.
Thank you so much for your time !
